### DISCUSSION TITLE
编译失败：ld: unknown option: --whole-archive

### DISCUSSION THREAD

**Discussion by TimmyLee** [alibaba/xquic#16] [Q&A] [answered]
ld: unknown option: --whole-archive
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libxquic.dylib] Error 1

**Comment by ruiqizhou** [comment]
Could you please provide more information, such as your system and gcc version?

**Reply by TimmyLee** [reply]
macOS , gcc:4.2.1

**Comment by Kulsk** [comment, marked as answer]
gcc with higher version is needed. We use gcc 4.8.5, and if you want to build xquic with the latest boringssl, gcc5 or gcc with higher version is needed.
If you will build xquic with the latest boringssl, gcc5 or gcc with higher version will be needed. 
After gcc is installed, you will get 2 gcc executable files in your system, one with path /usr/bin/gcc, the other with path /usr/local/bin/gcc if you configurated this directory to install gcc.

**Reply by TimmyLee** [reply]
Got it! I will try it later. Thanks
